### DISCUSSION TITLE
wire:navigate transitions triggered with right-click or middle-click

### DISCUSSION THREAD

**Discussion by guillaumelebon** [livewire/livewire#6077] [7. Bugs (archived)]
### Livewire version

3.0.0-beta.4

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

The wire:navigate feature is working really fine for me. Maybe a bit too much.

I've noticed an issue that I think is not the expected behavior.

When I **right-click** or **middle-click** on a link with wire:navigate, the transition is triggered. Typically, right-clicking or middle-clicking on a link is used to open the link in a new tab or to display the context menu, not to navigate to the link in the current tab.

I believe the expected behavior would be that wire:navigate is only triggered by a left-click (primary click), not a right-click or middle-click.

Best regards,
Guillaume

**Comment by joshhanley** [comment]
@guillaumelebon thanks for reporting! Any chance you could submit a failing test PR so we can investigate? Thanks!

**Comment by m7moudabdel7mid** [comment]
This is a workaround for this issue and also it prevents livewire from navigating when the user left clicks a link with ctrl key to open it in a new tab

replace `wire:navigate` with `@click="if(! event.metaKey) { event.preventDefault(); Alpine.navigate($el.getAttribute('href')); }`

> This workaround is not optimal since it will not prefetch the new page html on clickdown, and the request will start after the click is finished and mouseup is triggered.

**Comment by BoGnY** [comment]
same problem with 3.0.0-beta.5

`wire:navigate` isn't only triggered by a **left-click**

**Comment by joshhanley** [comment]
I have added a failing test PR #6266 for this issue.

**Comment by joshhanley** [comment]
PR #6266 has been fixed and merged. It will be in the next release 🙂

**Reply by m7moudabdel7mid** [reply]
That's great, Thank you, I think navigate shouldn't also trigger if the user holds meta key along with the left click, 
Instead the link should open in a new tab which is the default browser behaviour.

**Reply by guillaumelebon** [reply]
Thanks @joshhanley ! 🫡

I agree with @m7moudabdel7mid. Totally forgot about this case.

**Reply by apphaus-simon** [reply]
There seems to be another issue around this area - keyboard navigation with `wire:navigate` also seems to be prevented. Tabbing to a link and pressing enter doesn't `wire:navigate`, but also blocks regular page navigation.
